### PR TITLE
div. Changes

### PR DESCRIPTION
**Pull request by Moetez16**

### CODE REVIEW

File: server/gokbg3/grails-app/controllers/org/gokb/AjaxSupportController.groovy [hunk 1]
```diff
@@ -1666,4 +1666,54 @@ class AjaxSupportController {
 
     redirect(url: request.getHeader('referer'))
   }
+
+  @Transactional
+  @Secured(['ROLE_EDITOR', 'IS_AUTHENTICATED_FULLY'])
```
Comment (croqueGrec09) [inline server/gokbg3/grails-app/controllers/org/gokb/AjaxSupportController.groovy:1671]: ROLE_EDITOR should actually include IS_AUTHENTICATED_FULLY

File: server/gokbg3/grails-app/migrations/changelog-2021-04-15.groovy [hunk 2]
```diff
@@ -1,6 +1,234 @@
 databaseChangeLog = {
 
-    changeSet(author: "agalffy (hand-coded)", id: "1618484101333-1") {
+    changeSet(author: "djebeniani (generated)", id: "1618476753914-1") {
```
Comment (croqueGrec09) [inline server/gokbg3/grails-app/migrations/changelog-2021-04-15.groovy:3]: Duplicate Changeset-IDs ... please reorder!

File: server/gokbg3/grails-app/services/org/gokb/PackageService.groovy [hunk 3]
```diff
@@ -1693,7 +1673,7 @@ class PackageService {
     StringBuilder name = new StringBuilder()
     if (type == ExportType.KBART) {
       name.append(toCamelCase(pkg.provider?.name ? pkg.provider.name : "unknown Provider")).append('_')
-          .append(toCamelCase(pkg.scope.value)).append('_')
+          .append(toCamelCase(pkg.scope?.value ?: '')).append('_')
```
Comment (croqueGrec09) [inline server/gokbg3/grails-app/services/org/gokb/PackageService.groovy]: check presence of scope : pkg.scope ? pkg.scope.value : '', counts also for line above

File: server/gokbg3/grails-app/views/apptemplates/mainTemplates/_package.gsp [hunk 4]
```diff
@@ -117,6 +117,13 @@
         <li role="presentation"><a href="#altnames" data-toggle="tab">Alternate Names
           <span class="badge badge-warning"> ${d.variantNames?.size() ?: '0'}</span>
         </a></li>
+
+        <li>
+          <a href="#ddcs" data-toggle="tab">DDCs
+            <span class="badge badge-warning">${d.ddcs?.size() ?: '0'}</span>
```
Comment (croqueGrec09) [inline server/gokbg3/grails-app/views/apptemplates/mainTemplates/_package.gsp:123]: presence of lists should actually never be checked

File: server/gokbg3/grails-app/views/apptemplates/mainTemplates/_tipp.gsp [hunk 5]
```diff
@@ -262,6 +262,12 @@
             <span class="badge badge-warning">${d.prices?.size() ?: '0'}</span>
         </a>
     </li>
+
+    <li>
+        <a href="#ddcs" data-toggle="tab">DDCs
+            <span class="badge badge-warning">${d.ddcs?.size() ?: '0'}</span>
```
Comment (croqueGrec09) [inline server/gokbg3/grails-app/views/apptemplates/mainTemplates/_tipp.gsp:268]: see above

File: server/gokbg3/grails-app/views/apptemplates/secondTemplates/_nationalRange.gsp [hunk 6]
```diff
@@ -1,6 +1,6 @@
 <%@ page import="de.wekb.helper.RCConstants; org.gokb.cred.RefdataCategory" %>
 <ul>
-    <g:each in="${d.nationalRanges?.sort { it.value }}" var="nationalRange">
+    <g:each in="${d.nationalRanges?.sort { it.getI10n('value') }}" var="nationalRange">
```
Comment (croqueGrec09) [inline server/gokbg3/grails-app/views/apptemplates/secondTemplates/_nationalRange.gsp:3]: see above

File: server/gokbg3/grails-app/views/apptemplates/secondTemplates/_regionalRange.gsp [hunk 7]
```diff
@@ -1,6 +1,6 @@
 <%@ page import="de.wekb.helper.RCConstants; org.gokb.cred.RefdataCategory" %>
 <ul>
-    <g:each in="${d.regionalRanges?.sort { it.value }}" var="regionalRange">
+    <g:each in="${d.regionalRanges?.sort { it.getI10n('value') }}" var="regionalRange">
```
Comment (croqueGrec09) [inline server/gokbg3/grails-app/views/apptemplates/secondTemplates/_regionalRange.gsp:3]: see above

File: server/gokbg3/grails-app/views/public/_tippTabs.gsp [hunk 8]
```diff
@@ -39,6 +39,12 @@
             <span class="badge badge-warning">${d.prices?.size() ?: '0'}</span>
         </a>
     </li>
+
+    <li>
+        <a href="#ddcs" data-toggle="tab">DDCs
+            <span class="badge badge-warning">${d.ddcs?.size() ?: '0'}</span>
```
Comment (croqueGrec09) [inline server/gokbg3/grails-app/views/public/_tippTabs.gsp:45]: see above

File: server/gokbg3/grails-app/views/public/packageContent.gsp [hunk 9]
```diff
@@ -101,6 +101,13 @@
                         <span class="badge badge-warning">${pkg.variantNames?.size() ?: '0'}</span>
                     </a>
                 </li>
+
+                <li>
+                    <a href="#ddcs" data-toggle="tab">DDCs
+                        <span class="badge badge-warning">${pkg.ddcs?.size() ?: '0'}</span>
```
Comment (croqueGrec09) [inline server/gokbg3/grails-app/views/public/packageContent.gsp:107]: see above

File: server/gokbg3/grails-app/views/tabTemplates/_showDDCs.gsp [hunk 10]
```diff
@@ -0,0 +1,67 @@
+<%@ page import="de.wekb.helper.RCConstants; org.gokb.cred.RefdataCategory" %>
+<div class="tab-pane" id="ddcs">
+    <g:if test="${d.id != null}">
+        <dl>
+            <dt>
+                <gokb:annotatedLabel owner="${d}" property="ddcs">Dewey Decimal Classification</gokb:annotatedLabel>
+            </dt>
+            <dd>
+                <ul>
+                    <g:each in="${d.ddcs?.sort { it.value }}" var="ddc">
```
Comment (croqueGrec09) [inline server/gokbg3/grails-app/views/tabTemplates/_showDDCs.gsp:10]: see above

File: server/gokbg3/src/main/webapp/setup/DDC.csv [hunk 11]
```diff
@@ -0,0 +1,1001 @@
...
+DDC;7;Unbesetzt;Unassigned
+DDC;8;Unbesetzt;Unassigned
+DDC;9;Unbesetzt;Unassigned
+DDC;10;Bibliografie;Bibliography
+DDC;11;Bibliografien & Kataloge;Bibliographies
+DDC;12;Bibliografien & Kataloge von Einzelpersonen;Bibliographies of individuals
+DDC;13;Unbesetzt;Unassigned
+DDC;14;Anonymer & pseudonymer Werke;Of anonymous & pseudonymous works
+DDC;15;Von Werken aus einzelnen Orten;Bibliographies of works from specific places
+DDC;16;Von Werken �ber einzelne Themen;Bibliographies of works on specific subjects
```
Comment (croqueGrec09) [inline server/gokbg3/src/main/webapp/setup/DDC.csv:18]: Umlauts are not parsed correctly - is the charset correct?

File: server/gokbg3/src/main/webapp/setup/DDC_english.txt [hunk 12]
```diff
@@ -0,0 +1,1000 @@
+000;Computer science, information & general works;
```
Comment (croqueGrec09) [inline server/gokbg3/src/main/webapp/setup/DDC_english.txt:1]: is it wanted that the ending is txt and not csv?

File: server/gokbg3/src/main/webapp/setup/RefdataValue.csv [hunk 13]
```diff
@@ -0,0 +1,569 @@
...
+Country;AF;Afghanistan;Afghanistan
+Country;AG;Antigua und Barbuda;Antigua and Barbuda
+Country;AI;Anguilla;Anguilla
+Country;AL;Albanien;Albania
+Country;AM;Armenien;Armenia
+Country;AO;Angola;Angola
+Country;AQ;Antarktis;Antarctica
+Country;AR;Argentinien;Argentina
+Country;AS;Samoa, amerikanischer Teil;American Samoa
+Country;AT;�sterreich;Austria
```
Comment (croqueGrec09) [inline server/gokbg3/src/main/webapp/setup/RefdataValue.csv:61]: Umlauts not parsed as should

**Review by croqueGrec09 — COMMENTED** [review]
remarks

File: server/gokbg3/src/main/webapp/setup/DDC_english.txt (hunk 12 above)
Comment (Moetez16) [reply server/gokbg3/src/main/webapp/setup/DDC_english.txt:1]: Yes

File: server/gokbg3/grails-app/services/org/gokb/PackageService.groovy [hunk 14]
```diff
@@ -1693,7 +1673,7 @@ class PackageService {
     StringBuilder name = new StringBuilder()
     if (type == ExportType.KBART) {
       name.append(toCamelCase(pkg.provider?.name ? pkg.provider.name : "unknown Provider")).append('_')
```
Comment (croqueGrec09) [inline server/gokbg3/grails-app/services/org/gokb/PackageService.groovy]: Actually, I would expect here pkg.provider ? pkg.provider.name : "unknown provider"

File: server/gokbg3/src/main/webapp/setup/DDC.csv (hunk 11 above)
Comment (Moetez16) [reply server/gokbg3/src/main/webapp/setup/DDC.csv:18]: Yes

**Review by croqueGrec09 — APPROVED** [review]
approved

**Pull request merged**